jmclient/jmclient/wallet.py

```diff
@@ -1593,6 +1610,172 @@ def _get_mixdepth_from_path(self, path):
+    and is helpful for recovery of funds because the wallet can search
+    the only addresses corresponding to timenumbers which are far fewer
+
+    For example, if TIMENUMBER_UNIT = 1 (i.e. every time number is one month)
+    then there are 12 timelocks per year so just 1200 possible
+    addresses per century per pubkey. Easily searchable when recovering a
+    wallet from seed phrase. Therefore the user doesn't need to store any
+    dates, the seed phrase is sufficent for recovery.
+    """
+    #should be a factor of 12, the number of months in a year
+    TIMENUMBER_UNIT = 1
+
+    # all timelocks are 1st of the month at midnight
+    TIMELOCK_DAY_AND_SHORTER = (1, 0, 0, 0, 0)
+    TIMELOCK_EPOCH_YEAR = 2020
+    TIMELOCK_EPOCH_MONTH = 1 #january
+    MONTHS_IN_YEAR = 12
+
+    TIMELOCK_ERA_YEARS = 30
+    TIMENUMBERS_PER_PUBKEY = TIMELOCK_ERA_YEARS * MONTHS_IN_YEAR // TIMENUMBER_UNIT
+
+    """
+    As each pubkey corresponds to hundreds of addresses, to reduce load the
+    given gap limit will be reduced by this factor. Also these timelocked
+    addresses are never handed out to takers so there wont be a problem of
+    having many used addresses with no transactions on them.
+    """
+    TIMELOCK_GAP_LIMIT_REDUCTION_FACTOR = 6
+
+    #only one mixdepth will have fidelity bonds in it
+    FIDELITY_BOND_MIXDEPTH = 0
+
+    @classmethod
+    def _time_number_to_timestamp(cls, timenumber):
+        """
+        converts a time number to a datetime object
+        """
+        year = cls.TIMELOCK_EPOCH_YEAR + (timenumber*cls.TIMENUMBER_UNIT) // cls.MONTHS_IN_YEAR
+        month = cls.TIMELOCK_EPOCH_MONTH + (timenumber*cls.TIMENUMBER_UNIT) % cls.MONTHS_IN_YEAR
+        return datetime(year, month, *cls.TIMELOCK_DAY_AND_SHORTER)
+
+    @classmethod
+    def timestamp_to_time_number(cls, timestamp):
+        """
+        converts a datetime object to a time number
+        """
+        #dt = datetime.utcfromtimestamp(unixtimestamp)
+        dt = timestamp
+        if (dt.month - cls.TIMELOCK_EPOCH_MONTH) % cls.TIMENUMBER_UNIT != 0:
+            raise ValueError()
+        day_and_shorter_tuple = (dt.day, dt.hour, dt.minute, dt.second, dt.microsecond)
+        if day_and_shorter_tuple != cls.TIMELOCK_DAY_AND_SHORTER:
+            raise ValueError()
+        timenumber = (dt.year - cls.TIMELOCK_EPOCH_YEAR)*(cls.MONTHS_IN_YEAR //
+            cls.TIMENUMBER_UNIT) + ((dt.month - cls.TIMELOCK_EPOCH_MONTH) // cls.TIMENUMBER_UNIT)
+        if timenumber < 0 or timenumber > cls.TIMENUMBERS_PER_PUBKEY:
+            raise ValueError("datetime out of range")
+        return timenumber
+
+    @classmethod
+    def _is_timelocked_path(cls, path):
+        #return path[2] == cls.BIP32_TIMELOCK_ID and isinstance(path[-1], datetime)
+        return isinstance(path[-1], datetime)
+
+    @classmethod
+    def _is_burn_path(cls, path):
+        #TODO this doesnt work, because its hardened so the number is + 2*32 i think
```

Comment:
It's + 2**31 as per `BIP32Wallet._harden_path_level` (well as per BIP32, but code ref may be useful)

Reply:
TY. I'm working on adding burner outputs right now so it will help for figuring this out soon.